### ISSUE TITLE
02112023 react reports request data on resize

### ISSUE THREAD

**Issue by garsuga** [radiasoft/sirepo#5456]
Resizing the page causes reports in react to request their data from the API. This seems to be an indicator of a larger issue. Report elements are responsible for requesting their own data but this should not be the case. For manual reports this needs to be done centrally once the report is complete and the manual report elements should just retrieve this data from state.

**Comment by robnagler** [comment]
I think that manual and "automatic" reports should do the same thing: only update if the server says the data has changed. Otherwise, the state should be used always. All reports dependent on a computeJob should be invalidated (blank) when changes to the computeJob's parameters are made. This is not done on manual reports, but should be. Automatic reports, this is what happens. 

runSimulation should not return reports at all. We can change that with a flag that react passes and angular doesn't. This would simplify the code, because then automatic and manual would be the same.

**Comment by garsuga** [comment]
Yeah I agree, this issue is more about there being unwanted coupling between grabbing the report data from the server and rendering the report element in the app. The report data should be stored separately so that future renders grab existing data instead of requesting it again.

**Comment by garsuga** [comment]
I also need to redo the layout components a little at the same time, the waterfall layout need to "portal" components into new places instead of just creating them where it wants them. The current approach causes state to be lost during a layout.
